### PR TITLE
Omen markets details

### LINKED ISSUE #356: [UI/UX] Link to Omen markets, where relevant
From the condition details view, and potentially from other places in the app, we should be able to expose any omen markets that have been created for a position and link out to them.

### LINKED ISSUE #621: UI issues with 'Omen Market(s)' section on a condition/position details
Related to #595, #356

1. There is no 'copy' icon when there is 1 omen market in the section (it should be displayed according to the mock-up in the [comment](https://github.com/gnosis/conditional-tokens-explorer/issues/356#issuecomment-722363289) )

2.  'Navigate' icon is displayed in a separate line when an omen market text is long (it should be in the end of the word in the last line, I think)

### LINKED ISSUE #622: No question text for an Omen condition when there is no relevant Omen markets
related to #595, #356

See the pic: condition contains question ID and outcomes only.
However, there is no Question section with a question text

**Expected Result:** question text should be displayed?

ID (as an example): https://pr595--conditionaltokensexplorer.review.gnosisdev.com/#/conditions/0x065ebfcb8cd3e32033cfd42643578fb66a3bd6c664823b88fcf9d1c2dfb4c137

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #356 
Closes #621 
Closes #622  

An array is passed to a custom component to be used in any component but in his logic in only passes to the query the first one. 

I'm updating this because is just ready for review, I've found that a condition in the omen graph can be related to more than one market, for example, `http://localhost:3000/#/conditions/0x1e33c1fc428f35eeef468623bf773b20c6c24f591b491735f092b6410a2d5f95`, so I still need a way to integrate this with the question listed from another feature as @gabitoesmiapodo said.

From that condition, is possible to go to one of the positions and get listed all the markets of all the conditions related to that position (including the one used to find that position). 

Everything is flatted out, maybe if in the future we need to have a column for which condition is in the position, then we should change the data returned. Anyways more feedback should be addressed before getting to that point.

The `cag` graph for CTE is using the same contract as `protofire` graph for omen, so for listing markets I used the env vars in that way, but for linking to the `omen.eth.link` page the contract seems to be different because the same IDs are not listed. I think this can be solved only by putting the correct env vars.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr595--conditionaltokensexplorer.review.gnosisdev.com

**Review by gabitoesmiapodo — CHANGES_REQUESTED** [review]
I know the "design" says otherwise, but it would probably make more sense having the link to the market in the question for the Condition Details. Otherwise you'll have two items which are basically the same:

```
Question
Question text

Omen Market 
Question text [link]
```

I think it makes more sense having only:

```
Question
Question text [link]
```


Also: should this field be shown on the position details section? I'm asking because I couldn't find any.

In case it also belongs to the position details section, I think the `Omen Market: Question text [link]` format would be OK.

**Comment by hpmaxi** [comment]
@gabitoesmiapodo I think that is possible to have a position (from the list of positions of a condition with omen markets) that has a related condition with omen markets. But the problem is that 1 condition could be related to many markets, and a position with depth > 1 could have more than one condition (N), each with one or more markets (M), so is something like NxM markets listed in the position detail.

**Comment by gabitoesmiapodo** [comment]
@hpmaxi Yeah... I'm not sure I understood anything you said, but in case I did: if you can't put that info in a Position Details section for whatever reason (or if it's incredibly complicated), just don't.

I'm just asking because https://github.com/gnosis/conditional-tokens-explorer/issues/356 mentions something about positions ("(...) any omen markets that have been created for a position and link out to them.").

The issue about the Condition Details sections `Question / Omen Market` info still stands, though.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr595--conditionaltokensexplorer.review.gnosisdev.com

File: src/components/form/DisplayTableHashes/index.tsx [hunk 1]
```diff
@@ -20,17 +23,26 @@ export const DisplayTableHashes = (props: Props) => {
...
-          )
+        cell: ({ hash, title }: HashArray) => {
+          if (title) {
+            return (
+              <FlexRow>
+                {title}
+                <ExternalLink href={getOmenMarketURL(hash)} />
+              </FlexRow>
+            )
+          } else {
```
Comment (hpmaxi) [inline src/components/form/DisplayTableHashes/index.tsx:34]: This is the hacky part of the PR, I can't use the hash as a title but at the same time I need the hash to get the url.

File: src/hooks/useOmenMarkets.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+              hasCondition(data)
+            )
+            .map((data) => data.condition)
+            .filter((data) => data.fixedProductMarketMakers.length > 0) // [{ fpmm: [{ question, id }]}, {fpmm: ...} ]},
+            .map((data) => data.fixedProductMarketMakers) // [[{ question, id }, ..], [{question, id}, ...] ]},
+            .flat(1) // [ {question, id}, { question, id}]
+            .filter((data: OmenMarketFromTheGraph): data is ValidOmenMarket => hasTitle(data))
+            .map((market) => {
+              return { ...market, url: getOmenMarketURL(market.id) }
+            })
```
Comment (hpmaxi) [inline src/hooks/useOmenMarkets.tsx:68]: This has a lot of types and type guards to shape the data in a way more convenient for the consumer component.

**Review by mariano-aguero — APPROVED** [review]

**Comment by mariano-aguero** [comment]
@hpmaxi I give my approve, but let check gabriel feedback and the possible question in the channel

**Comment by hpmaxi** [comment]
@auryn-macmillan What do you think about the preview of this one, you can view it here `https://pr595--conditionaltokensexplorer.review.gnosisdev.com/#/conditions/0x1e33c1fc428f35eeef468623bf773b20c6c24f591b491735f092b6410a2d5f95`, and on the positions listed for that condition.

We have, only for conditions details, two sections very similar. One is the question that is only shown if it is an omen condition, that just interpret the `questionId` used in his creation as a Reality.eth entity, and the other is the `Omen Markets` section that search in the omen subgraph for conditions and his `fpmm`, and then extract the non-empty question `fpmm`.

We should try to unify this to only one section?

**Comment by auryn-macmillan** [comment]
> We should try to unify this to only one section?
Yes, I don't think there is any need to duplicate the question text.

For positions, it would be great to also show the text of the outcome(s) that correspond with the position.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr595--conditionaltokensexplorer.review.gnosisdev.com

**Review by gabitoesmiapodo — CHANGES_REQUESTED** [review]
It looks better but:

- We still should show the question when there are not markets for the condition.
- We should show something while the markets are being fetched. What we usually do is: Show the title (Omen Markets) and `-` for contents (or it could be `Loading...`). Once markets are fetched, replace the contents.

**Comment by gabitoesmiapodo** [comment]
@hpmaxi please also take a look at this https://app.zenhub.com/workspaces/conditional-tokens-explorer-5eecb9b75ab924001c8aa224/issues/gnosis/conditional-tokens-explorer/621

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr595--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr595--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr595--conditionaltokensexplorer.review.gnosisdev.com

**Review by gabitoesmiapodo — APPROVED** [review]

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr595--conditionaltokensexplorer.review.gnosisdev.com

**Comment by elena-zh** [comment]
Verified and LGTM (along with #621 and #622). 
Some more related UI issues are found and reported in #627

**Pull request merged**